scripts/redshift/loader.py

```diff
@@ -0,0 +1,155 @@
+import argparse
+import concurrent.futures
+import multiprocessing
+import os
+
+from matrix.common.aws.redshift_handler import RedshiftHandler
+from matrix.common.query_constructor import format_str_list
+from matrix.common.etl import (get_dss_client,
+                               etl_dss_bundles,
+                               transform_bundle,
+                               finalizer_reload,
+                               finalizer_update,
+                               upload_and_load,
+                               load_tables)
+
+# Match all SS2 and 10X analysis bundles
+DSS_SEARCH_QUERY_TEMPLATE = {
+    "query": {
+        "bool": {
+            "must": [
+                {
+                    "bool": {
+                        "should": [],
+                        "minimum_number_should_match": 1
+                    }
+                },
+                {
+                    "match": {
+                        "files.donor_organism_json.biomaterial_core.ncbi_taxon_id": 9606
+                    }
+                },
+                {
+                    "bool": {
+                        "should": [
+                            {
+                                "match": {
+                                    "files.analysis_process_json.type.text": "analysis"
+                                }
+                            },
+                            {
+                                "match": {
+                                    "files.analysis_process_json.process_type.text": "analysis"
+                                }
+                            }
+                        ],
+                        "minimum_number_should_match": 1
+                    }
+                }
+            ]
+        }
+    }
+}
+
+ALL_10X_SS2_MATCH_TERMS = [
+    {
+        "match": {
+            "files.library_preparation_protocol_json.library_construction_approach.ontology": "EFO:0008931"
+        }
+    },
+    {
+        "match": {
+            "files.library_preparation_protocol_json.library_construction_approach.ontology_label": "10X v2 sequencing"
+        }
+    },
+    {
+        "match": {
+            "files.library_preparation_protocol_json.library_construction_method.ontology": "EFO:0008931"
+        }
+    },
+    {
+        "match": {
+            "files.library_preparation_protocol_json.library_construction_method.ontology_label": "10X v2 sequencing"
+        }
+    }
+]
+
+
+def _build_dss_query(project_uuids):
+    q = DSS_SEARCH_QUERY_TEMPLATE
+
+    if project_uuids:
+        for uuid in project_uuids:
+            q['query']['bool']['must'][0]['bool']['should'].append({
```

Comment:
What does the nesting of `should` and `must` get you?

Reply:
Hm, it's enabling multiple `should` clauses (for which a should clause implements an OR to support multiple key names). The top level `must` is an array, as opposed to just a `bool` clause is a dict.